subprojects/logging/src/main/java/org/gradle/internal/logging/events/StyledTextOutputEvent.java

```diff
@@ -75,7 +85,11 @@ public String toString() {
     }
 
     public StyledTextOutputEvent withLogLevel(LogLevel logLevel) {
-        return new StyledTextOutputEvent(getTimestamp(), getCategory(), logLevel, spans);
+        return new StyledTextOutputEvent(getTimestamp(), getCategory(), logLevel, operationId, spans);
+    }
+
+    public OperationIdentifier getOperationId() {
```

Comment:
I think we could include this method in `RenderableOutputEvent` so other types of renderable events are also required to handle operation ids.